src/bin/pg_autoctl/monitor.c

```diff
@@ -346,6 +358,266 @@ monitor_node_active(Monitor *monitor,
+	char *value = NULL;
+	int errors = 0;
+
+	if (PQntuples(result) != 1)
+	{
+		log_error("Query returned %d rows, expected 1", PQntuples(result));
+		context->parsedOK = false;
+		return;
+	}
+
+	if (PQnfields(result) != 2)
+	{
+		log_error("Query returned %d columns, expected 2", PQnfields(result));
+		context->parsedOK = false;
+		return;
+	}
+
+	value = PQgetvalue(result, 0, 0);
+	if (sscanf(value, "%d", &context->candidatePriority) != 1)
+	{
+		log_error("Invalid failover candidate priority \"%s\" returned by monitor", value);
+		++errors;
+	}
+
+	value = PQgetvalue(result, 0, 1);
+	if (value == NULL || ( (*value != 't') && (*value != 'f')))
+	{
+		log_error("Invalid replication quorum \"%s\" returned by monitor", value);
+		++errors;
+	}
+	else
+	{
+		context->replicationQuorum = (*value) =='t';
+	}
+
+	if (errors > 0)
+	{
+		context->parsedOK = false;
+		return;
+	}
+
+	/* if we reach this line, then we're good. */
+	context->parsedOK = true;
+}
+
+
+/*
+ * monitor_get_formation_number_sync_standbys retrieves number-sync-stanbys
+ * property for formation from the monitor. The function returns true upon
+ * success.
+ */
+bool
+monitor_get_formation_number_sync_standbys(Monitor *monitor, char *formation,
+										   int *numberSyncStandbys)
+{
+	PGSQL *pgsql = &monitor->pgsql;
+	const char *sql = "SELECT number_sync_standbys FROM pgautofailover.formation "
+		"WHERE formationid = $1";
+	int paramCount = 1;
+	Oid paramTypes[1] = { TEXTOID};
+	const char *paramValues[1];
+	SingleValueResultContext parseContext = {PGSQL_RESULT_INT, false};
+	paramValues[0] = formation;
+
+	if (!pgsql_execute_with_params(pgsql, sql,
+								   paramCount, paramTypes, paramValues,
+								   &parseContext, parseSingleValueResult))
+	{
+		log_error("Failed retrieve settings of formation \"%s\".", formation);
+
+		/* disconnect from monitor */
+		pgsql_finish(&monitor->pgsql);
+
+		return false;
+	}
+
+	/* disconnect from monitor */
+	pgsql_finish(&monitor->pgsql);
+
+	if (!parseContext.parsedOk)
+	{
+		return false;
+	}
+
+	*numberSyncStandbys = parseContext.intVal;
+
+	return true;
+}
+
+
+/*
+ * monitor_set_formation_number_sync_standbys sets number-sync-stanbys
+ * property for formation at the monitor. The function returns true upon
+ * success.
+ */
+bool
+monitor_set_formation_number_sync_standbys(Monitor *monitor, char *formation,
+										   int numberSyncStandbys)
+{
+	PGSQL *pgsql = &monitor->pgsql;
+	const char *sql = "SELECT pgautofailover.set_formation_number_sync_stanbys($1, $2)";
+	int paramCount = 2;
+	Oid paramTypes[2] = { TEXTOID, INT4OID};
+	const char *paramValues[2];
+	SingleValueResultContext parseContext = {PGSQL_RESULT_BOOL, false};
+	paramValues[0] = formation;
+	paramValues[1] = intToString(numberSyncStandbys).strValue;
+
+	if (!pgsql_execute_with_params(pgsql, sql, paramCount, paramTypes, paramValues,
+								   &parseContext, parseSingleValueResult))
+	{
+		log_error("Failed retrieve settings of formation \"%s\".", formation);
```

Comment:
I think this is a copy-paste error, we are setting the parameter, not retrieving it.